### DISCUSSION TITLE
Is it possible to use zod to validate a comma-separated string

### DISCUSSION THREAD

**Discussion by grug** [colinhacks/zod#1869] [Q&A] [answered]
Hey there,

I'm trying to validate some queryparams coming through to an endpoint, i.e. `{ foo: '1,2,3,4' }`. I'd like to use zod to validate that the value of `foo` is a comma-separated string. Is anything like this possible?

**Comment by JacobWeisenburger** [comment]
what should the value of `foo` be?
- string array?
- number array?
- something else?

**Reply by grug** [reply]
Yep the second I read this I feel stupid 😂 probably doesn't make sense to try have this sort of validation as it's way too hairy.

**Reply by JacobWeisenburger** [reply]
there might be a way to do what you are asking. I'm just wanting to fully understand what you want before I start making suggestions.

**Reply by grug** [reply]
I _think_ in my example, I'd be looking at `foo` as having a value being a comma-separated string, where each comma-separated item passes validation for a number. I'm aware I could do this without zod, but I was just seeing if there's a mechanism available to do something like this.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const schema = z.object( {
    foo: z.string()
        .transform( value => value.split( ',' ).map( Number ) )
        .pipe( z.number().array() ),
} )

console.log( schema.parse( { foo: '1,2,3,4' } ) )
// { foo: [ 1, 2, 3, 4 ] }
```

**Reply by grug** [reply]
That is _exactly_ what I'm looking for.

Thanks @JacobWeisenburger - you are a legend.

**Reply by staciax** [reply]
thanks <3 it's work 

```ts

const schema = z.object({
	foo: z
		.string()
		.transform((value) => value.split(','))
		.pipe(z.string().trim().url().array()),
});

console.log(
	schema.parse({ foo: 'http://localhost,    http://localhost:8000' }),
);
```

```jsonc
{
  foo: [ "http://localhost", "http://localhost:8000" ],
}
```

**Reply by Montreal-Dev** [reply]
> thanks <3 it's work
> 
> ```ts
> const schema = z.object({
> 	foo: z
> 		.string()
> 		.transform((value) => value.split(','))
> 		.pipe(z.string().trim().url().array()),
> });
> 
> console.log(
> 	schema.parse({ foo: 'http://localhost,    http://localhost:8000' }),
> );
> ```
> 
> ```js
> {
>   foo: [ "http://localhost", "http://localhost:8000" ],
> }
> ```

My new favorite Zod pattern. Thank you!
